webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.test.js

```diff
@@ -1,7 +1,29 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from 'react-redux';
+import { Modal } from 'patternfly-react';
+import { mount } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
 import ForemanModal from '.';
 import { testComponentSnapshotsWithFixtures } from '../../common/testHelpers';
 
+const middlewares = [];
+const mockStore = configureMockStore(middlewares);
+const state = {
+  foremanModals: {
+    render: { open: true },
+    closed: { open: false },
+    withSuppliedChildren: { open: true },
+    unordered: { open: true },
+  },
+};
+
+const ConnectedForemanModal = props => (
```

Comment:
(Continuing from above)  So, how to write an Enzyme test for a component with that uses Hooks and react-redux?

There seems to be two options:

1. Mock the redux store and use Enzyme `mount`, making it more of an integration test.  This is what I had originally.
2. Mock `react-redux` including any Hooks used in the component.  That way, it allows you to use `shallow` which is closer to a unit test of the component, and it doesn't care about the Redux implementation.  This is referenced in the 2nd answer to [this StackOverflow question](https://stackoverflow.com/questions/56827300/how-to-test-a-component-using-react-redux-hooks).

I added a new test file, `ForemanModalNoRedux.test.js` that uses this implementation.  It does the following:
* Mocks `useSelector` with a fake store state
* Mocks `useDispatch` with a fake `dispatch` function
* All Redux-connected values are mocked, so we are just testing the component itself
* Use `testComponentSnapshotWithFixtures` for all tests, meaning `shallow` rendering is used and thus is a unit test
* Move the two tests I was using `mount` for into the fixtures and test them without deep rendering

In this way, I think we can keep the testing of Hooks components separate from testing Redux.  If this is acceptable, I will make this file the new `ForemanModal.test.js` and eliminate the old one.

@sharvit @waldenraines @laviro @MariaAga @johnpmitsch @jturel   anyone else, would also love to hear your thoughts!

Reply:
@amirfefer bumping the above comment :)

Reply:
IMO we can continue using the current approach.

- unit-testing for the `Component.js`
Where we import the `Component.js` file, using `shallow` and `snapshots` to make sure the component gets rendered correctly with each set of props.

- integration testing for the `index.js`
Where we import the `index.js` file, using `mount` to test that all the different parts are working correctly within the redux lifecycle.
`InitialState` -> `ComponentRendering` -> `EventTriggering` -> `ActionCalled` -> `StateUpdated` -> `ComponentRendering`

TBH I don't like the approach of importing the `index.js`, mocking everything inside so I can write a unit-test.
The unit is the component and I can import it directly from the `Component.js` file without the need to use the `index.js` file that needs many other moving parts to work together.

Reply:
> TBH I don't like the approach of importing the index.js, mocking everything inside so I can write a unit-test.

@sharvit Ok, I'm going to try redoing the tests again so that
* unit tests (`shallow` and your unit test helpers) apply to our (newly presentational) `ForemanModal.js`, and
* integration tests (`mount`) apply to our container component `index.js`.

Reply:
@sharvit @amirfefer Refactored the tests as stated above, ready for another look